File: FEBRUARY/06-02-2026/3634.java
class Solution {
    public int minRemoval(int[] nums, int k) {
        Arrays.sort(nums);
        int l = 0;
        int wlen = 0;
        for(int r = 0; r < nums.length; r++){
            while((long)nums[r] > (long)nums[l]*k){
                l++;
            }
            if(wlen < r-l+1){
                wlen = r-l+1;
            }
        }
        return nums.length - wlen;
    }
}

File: FEBRUARY/06-02-2026/3634.py
class Solution:
    def minRemoval(self, n: List[int], k: int) -> int:
        n.sort()
        o=l=0
        for r,v in enumerate(n):
            while k*n[l]<v:l+=1
            o=max(o,r-l)
        return len(n)-o-1      